<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
	 
	 <meta name="description" content="Modern Fortran Edition of the Pikaia Genetic Algorithm">
    
    <meta name="author" content="Chris MacMackin" >
    <link rel="icon" href="../favicon.png">

    <title>
All Procedures &ndash; Pikaia
</title>

    <link href="../css/bootstrap.min.css" rel="stylesheet">
    <link href="../css/pygments.css" rel="stylesheet">
    <link href="../css/font-awesome.min.css" rel="stylesheet">
    <link href="../css/local.css" rel="stylesheet">
    

    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    
    <script src="../js/jquery-2.1.3.min.js"></script>

    
    <script src="../tipuesearch/tipuesearch_content.js"></script>
    <link  href="../tipuesearch/tipuesearch.css" rel="stylesheet">
    <script src="../tipuesearch/tipuesearch_set.js"></script>
    <script src="../tipuesearch/tipuesearch.js"></script>
    

  </head>

  <body>

    <!-- Fixed navbar -->
    <nav class="navbar navbar-inverse navbar-fixed-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="../index.html">Pikaia </a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
				
                
            <li><a href="../lists/files.html">Source Files</a></li>
				
				
            <li><a href="../lists/modules.html">Modules</a></li>
				
				
            <li><a href="../lists/procedures.html">Procedures</a></li>
				
				
            <li><a href="../lists/absint.html">Abstract Interfaces</a></li>
								
            <li><a href="../lists/types.html">Derived Types</a></li>
				
				
            <li><a href="../program/pikaia_test.html">Program</a></li>
				
          </ul>
        
        <form action="../search.html" class="navbar-form navbar-right" role="search">
        <div class="form-group">
          <input type="text" class="form-control" placeholder="Search" name="q" id="tipue_search_input" autocomplete="off" required>
        </div>
<!--
        <button type="submit" class="btn btn-default">Submit</button>
-->
        </form>
        
        </div><!--/.nav-collapse -->
      </div>
    </nav>

    <div class="container">
    
      <div class="row">
        <div class="col-lg-12" id='text'>
			 <h1>Procedures</h1>
			 <table class="table table-striped">
			 <thead><tr><th>Procedure</th><th>Location</th><th>Procedure Type</th><th>Description</th></tr></thead>
			 <tbody>
			 
			   <tr><td><a href='../proc/adjmut.html'>adjmut</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Dynamical adjustment of mutation rate:</p></td></tr>
			 
			   <tr><td><a href='../proc/cross.html'>cross</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>breeds two parent chromosomes into two offspring chromosomes.
  breeding occurs through crossover. If the crossover probability
  test yields true (crossover taking place), either one-point or
  two-point crossover is used, with equal probabilities.</p></td></tr>
			 
			   <tr><td><a href='../proc/decode.html'>decode</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>decode genotype into phenotype parameters
  ph(k) are x,y coordinates [ 0 &lt; x,y &lt; 1 ]</p></td></tr>
			 
			   <tr><td><a href='../proc/encode.html'>encode</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Encode phenotype parameters into integer genotype
  ph(k) are x,y coordinates [ 0 &lt; x,y &lt; 1 ]</p></td></tr>
			 
			   <tr><td><a href='../proc/func_wrapper.html'>func_wrapper</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Wrapper for the user's function that is used by the main pikaia routine
  The x input to this function comes from pikaia, and will be between [0,1].</p></td></tr>
			 
			   <tr><td><a href='../proc/genrep.html'>genrep</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Full generational replacement: accumulate offspring into new
  population array</p></td></tr>
			 
			   <tr><td><a href='../proc/mutate.html'>mutate</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Introduces random mutation in a genotype.
  Mutations occur at rate pmut at all gene loci.</p></td></tr>
			 
			   <tr><td><a href='../proc/newpop.html'>newpop</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Replaces old population by new; recomputes fitnesses &amp; ranks</p></td></tr>
			 
			   <tr><td><a href='../proc/pikaia.html'>pikaia</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Optimization (maximization) of user-supplied "fitness" function
  over n-dimensional parameter space x using a basic genetic
  algorithm method.</p></td></tr>
			 
			   <tr><td><a href='../proc/report.html'>report</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Write generation report to standard output</p></td></tr>
			 
			   <tr><td><a href='../proc/report_iteration.html'>report_iteration</a></td><td><a href='../program/pikaia_test.html'>pikaia_test</a></td><td>Subroutine</td><td><p>A simple iteration reporting function.
 Writes iter,x,f to the output file.</p></td></tr>
			 
			   <tr><td><a href='../proc/rninit.html'>rninit</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Initialize the random number generator with the input seed value.</p></td></tr>
			 
			   <tr><td><a href='../proc/rnkpop.html'>rnkpop</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Ranks initial population.
  Calls external sort routine to produce key index and rank order
  of input array arrin (which is not altered).</p></td></tr>
			 
			   <tr><td><a href='../proc/rosenbrock.html'>rosenbrock</a></td><td><a href='../program/pikaia_test.html'>pikaia_test</a></td><td>Subroutine</td><td><p>Rosenbrock function for testing the algorithm.
 The minimum is at f(1,1) = 0.
 http://en.wikipedia.org/wiki/Rosenbrock_function</p></td></tr>
			 
			   <tr><td><a href='../proc/rqsort.html'>rqsort</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Return integer array p which indexes array a in increasing order.
  Array a is not disturbed.  The Quicksort algorithm is used.</p></td></tr>
			 
			   <tr><td><a href='../proc/select_parents.html'>select_parents</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Selects two parents from the population, using roulette wheel
  algorithm with the relative fitnesses of the phenotypes as
  the "hit" probabilities.</p></td></tr>
			 
			   <tr><td><a href='../proc/set_inputs.html'>set_inputs</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Constructor for the <a href="../type/pikaia_class.html">pikaia_class</a>.
  The routine must be called before the solve routine can be used.</p></td></tr>
			 
			   <tr><td><a href='../proc/solve_with_pikaia.html'>solve_with_pikaia</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Main pikaia wrapper used by the class.</p></td></tr>
			 
			   <tr><td><a href='../proc/stdrep.html'>stdrep</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Subroutine</td><td><p>Steady-state reproduction: insert offspring pair into population
  only if they are fit enough (replace-random if irep=2 or
  replace-worst if irep=3).</p></td></tr>
			 
			   <tr><td><a href='../proc/twod.html'>twod</a></td><td><a href='../program/pikaia_test.html'>pikaia_test</a></td><td>Subroutine</td><td><p>Compute sample fitness function 
 (a smooth 2-d landscape)</p></td></tr>
			 
			   <tr><td><a href='../proc/urand.html'>urand</a></td><td><a href='../module/pikaia_module.html'>pikaia_module</a></td><td>Function</td><td><p>Return the next pseudo-random deviate from a sequence which is
  uniformly distributed in the interval [0,1]</p></td></tr>
			 
			 </tbody></table>
        </div>
      </div>

    <hr>
    <footer>
        <div class="row">
        <div class="col-md-6">&copy; 2015 Pikaia was written by Jacob Williams. </div>
        <div class="col-md-6"><span class="pull-right">Documentation generated by <a href="https://github.com/cmacmackin/ford">FORD</a>.</span></div>
        </div>
        <br>
    </footer>
    </div> <!-- /container -->


    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
<!--
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
-->
    <script src="../js/bootstrap.min.js"></script>
    <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
    <script src="../js/ie10-viewport-bug-workaround.js"></script>

    <!-- MathJax JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        TeX: { extensions: ['AMSmath.js','AMSsymbols.js','noErrors.js','noUndefined.js'], equationNumbers: { autoNumber: 'AMS' } },
        jax: ['input/TeX','input/MathML','output/HTML-CSS'],
        extensions: ['tex2jax.js','mml2jax.js','MathMenu.js','MathZoom.js'],
        'HTML-CSS': { 
           styles: { '.MathJax_Display, .MathJax .mo, .MathJax .mi, .MathJax .mn': {color: '#000000 ! important'} }
        }
      });
    </script>
    <script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  </body>
</html>